### ISSUE TITLE
Support "prefer async" in agent push

### ISSUE THREAD

**Issue by codyebberson** [medplum/medplum#4106]

**Comment by jmalobicky** [comment]
Is this based off https://hl7.org/fhir/r4/async.html ? with the `Prefer: respond-async` Header usage

**Comment by rahul1** [comment]
## Use Case
User has a Bot that pushes to an on-premise device via the Agent. The on premise device may take a long time to respond, and the Bot should return control back to the client application in the meantime, so as to not block the client

**Comment by jmalobicky** [comment]
Hello @ThatOneBro, This should be on the Bot not the Agent. In my use case it makes sense to have the pushToAgent blocking until I can get an MSA segment response. It is the call to the Bot that needs to have the `Prefer: respond-async` to not block the calling app while waiting on the MSA segment response.

**Comment by ThatOneBro** [comment]
Hey @jmalobicky, sorry about that. I think your initial request was lost in translation. Thanks for clarifying 

@rahul1 opened a new issue (#4458) to track enabling `Prefer-async` on the `Bot/$execute` operation, which should enable the workflow you're describing

**Comment by ThatOneBro** [comment]
Hey @jmalobicky, this will be live (probably tomorrow) in [3.1.5](https://github.com/medplum/medplum/pull/4481). Let me know if you have any more issues with this. Closing this one out
